web/src/utils/filter-definitions.ts

```diff
@@ -134,435 +124,192 @@ const isOwnerKind = (kind: string) => {
-        encoder: simpleFiltersEncoder('TimeFlowRttNs')
+  const protoValidation = (value: string) => {
+    if (_.isEmpty(value)) {
+      return invalid(t('Value is empty'));
+    }
+    //allow any protocol number or valid name / value
+    if (value == undefinedValue || !isNaN(Number(value))) {
+      return valid(value);
+    } else {
+      const proto = findProtocolOption(value);
+      if (proto) {
+        return valid(proto.name);
       }
-    ];
-  }
+      return invalid(t('Unknown protocol'));
+    }
+  };
 
-  if (allowConnectionFilter && allowDNSFilter && allowPktDrops && allowRTTFilter) {
-    return filterDefinitions;
-  } else {
-    return filterDefinitions.filter(
-      fd =>
-        (allowConnectionFilter || fd.id !== 'id') &&
-        (allowDNSFilter || !fd.id.startsWith('dns_')) &&
-        (allowPktDrops || !fd.id.startsWith('pkt_drop_')) &&
-        (allowRTTFilter || fd.id !== 'time_flow_rtt')
-    );
-  }
+  const dirValidation = (value: string) => {
+    if (_.isEmpty(value)) {
+      return invalid(t('Value is empty'));
+    }
+    //allow 0 / 1 or Ingress / Egress
+    const found = findDirectionOption(value, t);
+    if (found) {
+      return valid(found.name);
+    }
+    return invalid(t('Unknown direction'));
+  };
+
+  const getFuncs = (d: FilterConfigDef) => {
+    const isSrc = d.id.includes('src');
+    const colConfig = columnsDefs.find(c => c.filter === d.id);
+
+    let getOptions: (value: string) => Promise<FilterOption[]> = noOption;
+    let validate: (value: string) => { val?: string; err?: string } = rejectEmptyValue;
+    let encoder: FiltersEncoder = simpleFiltersEncoder(colConfig?.field as Field);
+    let checkCompletion:
+      | ((value: string, selected: string) => { completed: boolean; option: FilterOption })
+      | undefined = undefined;
+
+    if (d.id.includes('namespace')) {
+      getOptions = cap10(getNamespaceOptions);
+      validate = k8sNameValidation;
+    } else if (d.id.includes('name')) {
+      validate = k8sNameValidation;
+    } else if (d.id.includes('kind')) {
+      getOptions = cap10(getKindOptions);
+      validate = rejectEmptyValue;
+      encoder = kindFiltersEncoder(`${isSrc ? 'Src' : 'Dst'}K8S_Type`, `${isSrc ? 'Src' : 'Dst'}K8S_OwnerType`);
+    } else if (d.id.includes('resource')) {
+      getOptions = cap10(getResourceOptions);
+      validate = k8sResourceValidation;
+      checkCompletion = k8sResourceCompletion;
+      encoder = k8sResourceFiltersEncoder(
+        `${isSrc ? 'Src' : 'Dst'}K8S_Type`,
+        `${isSrc ? 'Src' : 'Dst'}K8S_OwnerType`,
+        `${isSrc ? 'Src' : 'Dst'}K8S_Namespace`,
+        `${isSrc ? 'Src' : 'Dst'}K8S_Name`,
+        `${isSrc ? 'Src' : 'Dst'}K8S_OwnerName`
+      );
+    } else if (d.id.includes('address')) {
+      validate = addressValidation;
+    } else if (d.id.includes('port')) {
+      getOptions = cap10(getPortOptions);
+      validate = portValidation;
```

Comment:
I wonder if, rather than that, it would be better to name the bunch of "validators" defined above (ie. the validation functions) and refer to these names in the config ? That would be more in line with the desired config-driven approach ?

Reply:
well, I admit that things like that make it slighlty more complicated: 
```
      encoder = k8sResourceFiltersEncoder(
        `${isSrc ? 'Src' : 'Dst'}K8S_Type`,
        `${isSrc ? 'Src' : 'Dst'}K8S_OwnerType`,
        `${isSrc ? 'Src' : 'Dst'}K8S_Namespace`,
        `${isSrc ? 'Src' : 'Dst'}K8S_Name`,
        `${isSrc ? 'Src' : 'Dst'}K8S_OwnerName`
      );
```

Reply:
I agree. As same as columns, we need to find a tradeoff between config and hardcoded switches. 
For now I choosed a simple approach for simplicity to introduce the config without breaking changes.

Ideally, I would like to [have a way to write predefined javascript functions](https://stackoverflow.com/questions/36517173/how-to-store-a-javascript-function-in-json) as same as `calculated` field from columns with more capabilities.